sdk/internal/testframework/recording_sanitizer.go

```diff
@@ -0,0 +1,85 @@
+// +build go1.13
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package testframework
+
+import (
+	"net/http"
+
+	"github.com/dnaeon/go-vcr/cassette"
+	"github.com/dnaeon/go-vcr/recorder"
+)
+
+type RecordingSanitizer struct {
+	recorder          *recorder.Recorder
+	headersToSanitize map[string]*string
+	urlSanitizer      StringSanitizer
+	bodySanitizer     StringSanitizer
+}
+
+type StringSanitizer func(*string)
```

Comment:
Is there a reason why this is pointer-to-string, i.e. does the empty string have semantic importance?

Reply:
There may not be a good reason for this to be a pointer.

Reply:
Thought about this more, I think it needs to be a pointer so that we can change the value passed in.

Reply:
Ah I see, I didn't realize this is used as an out parameter.  A comment for this type would be good.

Reply:
(Probably because out params are somewhat uncommon in Go)